cupyx/fallback_mode/fallback.py

```diff
@@ -76,12 +100,239 @@ def __call__(self, *args, **kwargs):
         if self._cupy_object is not None:
-            return utils._call_cupy(self._cupy_object, args, kwargs)
+            return _call_cupy(self._cupy_object, args, kwargs)
 
         # Execute numpy method
-        return utils._call_numpy(self._numpy_object, args, kwargs)
+        return _call_numpy(self._numpy_object, args, kwargs)
 
 
 numpy = _RecursiveAttr(np, cp)
+
+# -----------------------------------------------------------------------------
+# utils
+# -----------------------------------------------------------------------------
+
+
+def _call_cupy(func, args, kwargs):
+    """
+    Calls cupy function with *args and **kwargs and
+    does necessary data transfers.
+
+    Args:
+        func: A cupy function that needs to be called.
+        args (tuple): Arguments.
+        kwargs (dict): Keyword arguments.
+
+    Returns:
+        Result after calling func and performing data transfers.
+    """
+
+    args, kwargs = _get_cupy_args(args, kwargs)
+
+    res = func(*args, **kwargs)
+
+    return _get_fallback_result(res)
+
+
+def _call_numpy(func, args, kwargs):
+    """
+    Calls numpy function with *args and **kwargs and
+    does necessary data transfers.
+
+    Args:
+        func: A numpy function that needs to be called.
+        args (tuple): Arguments.
+        kwargs (dict): Keyword arguments.
+
+    Returns:
+        Result after calling func and performing data transfers.
+    """
+
+    args, kwargs = _get_cupy_args(args, kwargs)
+
+    numpy_args, numpy_kwargs = _get_numpy_args(args, kwargs)
+
+    numpy_res = func(*numpy_args, **numpy_kwargs)
+
+    cupy_res = _get_cupy_result(numpy_res)
+
+    return _get_fallback_result(cupy_res)
+
+
+# -----------------------------------------------------------------------------
+# ndarray wrapper and proxy magic methods
+# -----------------------------------------------------------------------------
+
+
+class ndarray:
+    """
+    Wrapper around cupy.ndarray
+    Gets initialized with a cupy ndarray.
+    """
+
+    def __init__(self, array):
+        self._array = array
+
+    def __getattr__(self, attr):
+        """
+        Catches attributes corresponding to ndarray.
+
+        Args:
+            attr (str): Attribute of ndarray class.
+
+        Returns:
+            (_RecursiveAttr object, self._array.attr):
+            Returns_RecursiveAttr object with numpy_object, cupy_object.
+            Returns self._array.attr if attr is not callable.
+        """
+
+        cupy_object = getattr(cp.ndarray, attr, None)
+
+        numpy_object = getattr(np.ndarray, attr)
+
+        if not callable(numpy_object):
+            return getattr(self._array, attr)
+
+        return _RecursiveAttr(numpy_object, cupy_object, self)
+
+    def _get_array(self):
+        """
+        Returns _array (cupy.ndarray) of ndarray object.
+        """
+        return self._array
+
+
+# Decorator for ndarray magic methods
+def make_method(name):
+    def method(self, *args, **kwargs):
+
```

Comment:
Why not erasing the newlines in this function? I think they do not improve readability.
```suggestion
```

Reply:
Done